File: csharp/src/SCF/SCFWriter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace Serengetti.SCF {
    public class SCFWriter {
        private Stream stream;
        private StreamWriter writer;
        public SCFWriter(Stream stream) {
            this.stream = stream;
            StreamWriter writer = new StreamWriter(stream);
        }
        public SCFWriter(String path) {
            if (!File.Exists(path)) {
                throw new FileNotFoundException("SCFWriter: File " + path + " could not be found.");
            }
            stream = new FileStream(path, FileMode.Create);
            writer = new StreamWriter(stream);
        }
        public void close() {
            writer.Close();
        }
        public void write(Dictionary<String, String> nameValue) {
            foreach (KeyValuePair<String, String> pair in nameValue) {
                writer.WriteLine(pair.Key + ":" + pair.Value);
            }
        }
    }
}

File: java/src/src/com/serengetti_software/scf/SCFWriter.java
package com.serengetti_software.scf;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.OutputStream;
import java.io.OutputStreamWriter;
import java.util.HashMap;
import java.util.Map;

public class SCFWriter {
	private OutputStreamWriter writer;
	public SCFWriter(OutputStream stream) {
		this.writer = new OutputStreamWriter(stream);
	}
	public SCFWriter(String path) throws FileNotFoundException {
		this(new FileOutputStream(path));
	}
	public void close() throws IOException {
		writer.close();
	}
	public void write(HashMap<String, String> nameValue) throws IOException {
		for(Map.Entry<String, String> pair : nameValue.entrySet()) {
			System.out.println("Writing: " + pair.getKey() + ":" + pair.getValue() + "\n");
			writer.write(pair.getKey() + ":" + pair.getValue() + "\n");
			writer.flush();
		}
	}
}
